### DISCUSSION TITLE
[StimulusBundle] Event listener priority between Stimulus controllers

### DISCUSSION THREAD

**Discussion by thomas-keck** [symfony/ux#2912] [Questions & Answers]
Hello everyone,

I need advice about Stimulus controllers events and listeners.
I have sometimes priority issues with event listeners.

The password input using Symfony UX Toggle Password and my custom controller :

```
<input
    type="password"
    id="password"
    name="_password"
    class="form-control"
    placeholder="Mot de passe"
    data-controller="toggle-password symfony--ux-toggle-password--toggle-password" data-symfony--ux-toggle-password--toggle-password-button-classes-value="&#x5B;&quot;btn&quot;,&quot;btn-lg&quot;,&quot;btn-outline-default&quot;&#x5D;" data-symfony--ux-toggle-password--toggle-password-visible-label-value="" data-symfony--ux-toggle-password--toggle-password-hidden-label-value=""
>

```

I need to append controller’s element to closest `div.input-group`. Here is my controller :

```
import { Controller } from "@hotwired/stimulus";

/* stimulusFetch: 'lazy' */
export default class extends Controller {
    connect() {
        this.element.addEventListener('toggle-password:connect', this._onConnect);
    }

    disconnect() {
        this.element.removeEventListener('toggle-password:connect', this._onConnect);
    }

    _onConnect(event) {
        event.detail.element.closest('.input-group').appendChild(event.detail.button);
    }
}
```

Sometimes, Symfony UX Toggle Password controller launches first, dispatchs the `toggle-password:connect` and then my controller is executed.

Because of this, my controller misses Symfony UX Toggle Password controller’s connect event and don’t update element position in the DOM.

Is there a way to add a “priority” between controller’s events on a single DOM element to make sure event listeners work as intended ?

Thank you for your support !

**Comment by smnandre** [comment]
Did you check the network tab of your browser "debug tools"  ?  

As this one is registered lazy, **maybe** here even if the order is the correct one, the time for the Stimulus lazy loader to fetch/load the controller file the other one has connected.

Could you try without stimulusFetch: lazy to see ?

**Comment by smnandre** [comment]
But i wonder, re-reading your issue, if the best way to achieve your goal would not be to use a custom form theme (or just disabling the one provided) ? 

See [Usage without Symfony Forms](https://symfony.com/bundles/ux-toggle-password/current/index.html#usage-without-symfony-forms) and [Usage with Symfony Forms](https://symfony.com/bundles/ux-toggle-password/current/index.html#usage-with-symfony-forms)

**Comment by thomas-keck** [comment]
Hello @smnandre,
Thank you for your support.

After some tests without stimulusFetch: lazy it seems the bug don't appear again.

I have the same issue on other elements with multiple Stimulus controllers.

I close this issue for now but I leave my ticket open on [Hotwire Discussion](https://discuss.hotwired.dev/t/event-listener-priority-between-controllers/5952)
